### PR TITLE
mgmt, support hibernation for OS on the disk

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

support https://github.com/Azure/azure-sdk-for-java/issues/32375

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/DiskImpl.java [hunk 1]
```diff
@@ -137,6 +137,11 @@ public Mono<Void> revokeAccessAsync() {
         return this.manager().serviceClient().getDisks().revokeAccessAsync(this.resourceGroupName(), this.name());
     }
 
+    @Override
+    public boolean supportsHibernation() {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/DiskImpl.java:141]: not sure if `isHibernationSupported` be better? hard to tell by me.

**Review by weidongxu-microsoft — APPROVED** [review]
@haolingdong-msft  for review as well.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/DiskImpl.java (hunk 1 above)
Comment (haolingdong-msft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/DiskImpl.java:141]: https://github.com/Azure/azure-sdk-for-java/blob/d1fbfcb0804ed2eb848ca79a00128d4d708016eb/sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java#L31
Here we also use `isAttachedToVirtualMachine`. Maybe `isHibernationSupported ` is better?

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/DiskImpl.java:141]: Yes, `isHibernationSupported` is better to align with current mgmt pattern for boolean, thanks!

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**